### PR TITLE
Some cleanups

### PR DESCRIPTION
**Pull request by heckflosse**

### CODE REVIEW

File: rtgui/extprog.cc [hunk 1]
```diff
@@ -137,13 +137,11 @@ bool ExtProgStore::searchProgram (const Glib::ustring& name,
     action.name = name;
     action.target = (allowRaw ? 1 : 2);
 
-    auto& filePath = action.filePathEXE;
```
Comment (Floessie) [inline rtgui/extprog.cc]: This is later checked. Mind the `&`.

File: rtengine/improcfun.cc [hunk 2]
```diff
@@ -5381,36 +5389,32 @@ void ImProcFunctions::getAutoExp  (const LUTu &histogram, int histcompr, double
...
 
-    for (; i < imax; i++) {
+    for (; j < imax; ++j) {
         if (count < 8) {
-            octile[count] += histogram[i];
+            octile[count] += histogram[j];
 
             if (octile[count] > sum / 8.f || (count == 7 && octile[count] > sum / 16.f)) {
-                octile[count] = xlog (1. + (float)i) / log (2.f);
+                octile[count] = xlog (1. + (float)j) / log (2.f);
```
Comment (Floessie) [inline rtengine/improcfun.cc]: static_cast.

File: rtgui/extprog.cc [hunk 3]
```diff
@@ -279,9 +272,9 @@ bool ExtProgStore::openInGimp (const Glib::ustring& fileName)
     for (auto ver = 12; ver >= 0; --ver) {
 
         executable = Glib::build_filename (options.gimpDir, "bin", Glib::ustring::compose (Glib::ustring("gimp-2.%1.exe"), ver));
-        auto success = ShellExecute( NULL, "open", executable.c_str(), fileName.c_str(), NULL, SW_SHOWNORMAL );
+        auto lsuccess = ShellExecute( NULL, "open", executable.c_str(), fileName.c_str(), NULL, SW_SHOWNORMAL );
 
-        if ((uintptr_t)success > 32) {
+        if ((uintptr_t)lsuccess > 32) {
```
Comment (Floessie) [inline rtgui/extprog.cc]: static_cast

File: rtgui/filebrowser.cc [hunk 4]
```diff
@@ -114,16 +114,16 @@ void findOriginalEntries (const std::vector<ThumbBrowserEntryBase*>& entries)
 
     // Find the original image for each bucket
     for (BasenameIterator bucket = byBasename.begin (); bucket != byBasename.end (); ++bucket) {
-        const EntryVector& entries = bucket->second;
+        const EntryVector& lentries = bucket->second;
         ThumbBrowserEntryBase* original = nullptr;
 
         // Select the most likely original in a first pass...
-        for (EntryIterator entry = entries.begin (); entry != entries.end (); ++entry) {
+        for (EntryIterator entry = lentries.begin (); entry != lentries.end (); ++entry) {
```
Comment (Floessie) [inline rtgui/filebrowser.cc]: Range-based for loop?

File: rtgui/guiutils.cc [hunk 5]
```diff
@@ -946,22 +946,21 @@ bool MyScrolledWindow::on_scroll_event (GdkEventScroll* event)
         const double lowerBound = adjust->get_lower();
         double value = adjust->get_value();
         double step  = adjust->get_step_increment();
-        double value2 = 0.;
 
         if (event->direction == GDK_SCROLL_DOWN) {
-            value2 = rtengine::min<double>(value + step, upperBound);
+            double value2 = rtengine::min<double>(value + step, upperBound);
```
Comment (Floessie) [inline rtgui/guiutils.cc]: const

File: rtgui/lockablecolorpicker.h [hunk 6]
```diff
@@ -74,14 +74,14 @@ class LockableColorPicker : BackBuffer
...
 
-    void draw (Cairo::RefPtr<Cairo::Context> &cr);
+    void draw (const Cairo::RefPtr<Cairo::Context> &cr);
 
     // Used to update the RGB color, the HSV values will be updated accordingly
     void setPosition (const rtengine::Coord &newPos);
     void setRGB (const float R, const float G, const float B, const float previewR, const float previewG, const float previewB);
-    void getImagePosition (rtengine::Coord &imgPos);
+    void getImagePosition (rtengine::Coord &imgPos) const;
     void getScreenPosition (rtengine::Coord &screenPos);
```
Comment (Floessie) [inline rtgui/lockablecolorpicker.h]: const?

File: rtgui/progressconnector.h [hunk 7]
```diff
@@ -79,9 +79,9 @@ class ProgressConnector
     static int emitEndSignalUI (void* data)
     {
 
-        sigc::signal0<bool>* opEnd = (sigc::signal0<bool>*) data;
-        int r = opEnd->emit ();
-        delete opEnd;
+        sigc::signal0<bool>* lopEnd = (sigc::signal0<bool>*) data;
```
Comment (Floessie) [inline rtgui/progressconnector.h]: reinterpret_cast

File: rtgui/progressconnector.h [hunk 8]
```diff
@@ -79,9 +79,9 @@ class ProgressConnector
...
+        int r = lopEnd->emit ();
```
Comment (Floessie) [inline rtgui/progressconnector.h]: const

File: rtgui/rtwindow.cc [hunk 9]
```diff
@@ -44,14 +44,14 @@ extern unsigned char initialGdkScale;
 static gboolean
 osx_should_quit_cb (GtkosxApplication *app, gpointer data)
 {
-    RTWindow *rtWin = (RTWindow *)data;
+    RTWindow *rtWin = static_cast<RTWindow*>(data);
```
Comment (Floessie) [inline rtgui/rtwindow.cc]: `* const`.

File: rtgui/thumbnail.cc [hunk 10]
```diff
@@ -590,10 +590,8 @@ void Thumbnail::decreaseRef ()
     cachemgr->closeThumbnail (this);
 }
 
-void Thumbnail::getThumbnailSize (int &w, int &h, const rtengine::procparams::ProcParams *pparams)
+int Thumbnail::getThumbnailWidth (const int &h, const rtengine::procparams::ProcParams *pparams) const
```
Comment (Floessie) [inline rtgui/thumbnail.cc]: `const int&`?

**Review by Floessie — CHANGES_REQUESTED** [review]
Just a quick sweep. One potential bug detected. Otherwise fine. :+1:

File: rtgui/thumbnail.cc (hunk 10 above)
Comment (heckflosse) [reply rtgui/thumbnail.cc]: Done

File: rtgui/progressconnector.h (hunk 8 above)
Comment (heckflosse) [reply rtgui/progressconnector.h]: Done

File: rtgui/lockablecolorpicker.h (hunk 6 above)
Comment (heckflosse) [reply rtgui/lockablecolorpicker.h]: Done

File: rtgui/guiutils.cc (hunk 5 above)
Comment (heckflosse) [reply rtgui/guiutils.cc]: Done

File: rtgui/filebrowser.cc (hunk 4 above)
Comment (heckflosse) [reply rtgui/filebrowser.cc]: Done

File: rtengine/improcfun.cc (hunk 2 above)
Comment (heckflosse) [reply rtengine/improcfun.cc]: cast removed

File: rtgui/extprog.cc (hunk 3 above)
Comment (heckflosse) [reply rtgui/extprog.cc]: That gives
```
extprog.cc:277:44: error: invalid static_cast from type 'HINSTANCE__*' to type 'uintptr_t' {aka 'long long unsigned int'}
  277 |         if (static_cast<uintptr_t>(lsuccess) > 32) {
```

File: rtgui/guiutils.cc (hunk 5 above)
Comment (heckflosse) [reply rtgui/guiutils.cc]: Done

File: rtgui/lockablecolorpicker.h (hunk 6 above)
Comment (heckflosse) [reply rtgui/lockablecolorpicker.h]: Done

File: rtgui/progressconnector.h (hunk 7 above)
Comment (heckflosse) [reply rtgui/progressconnector.h]: Done

File: rtgui/progressconnector.h (hunk 8 above)
Comment (heckflosse) [reply rtgui/progressconnector.h]: Done

File: rtgui/thumbnail.cc (hunk 10 above)
Comment (heckflosse) [reply rtgui/thumbnail.cc]: Done

File: rtgui/rtwindow.cc (hunk 9 above)
Comment (heckflosse) [reply rtgui/rtwindow.cc]: Done

File: rtgui/extprog.cc (hunk 1 above)
Comment (heckflosse) [reply rtgui/extprog.cc]: Fixed

File: rtgui/extprog.cc (hunk 3 above)
Comment (Floessie) [reply rtgui/extprog.cc]: That smells, doesn't it. But if that's the Win-way of doing it, then use a `reinterpret_cast`...

**Pull request merged**